src/python/gudhi/clustering/tomato.py

```diff
@@ -257,22 +254,24 @@ def fit(self, X, y=None, weights=None):
 
         self.weights_ = weights
         # This is where the main computation happens
-        self.leaf_labels_, self.children_, self.diagram_, self.max_weight_per_cc_ = hierarchy(
+        self.leaf_labels_, self.children_, self.diagram_, self.max_weight_per_cc_ = _hierarchy(
             self.neighbors_, weights
         )
         self.n_leaves_ = len(self.max_weight_per_cc_) + len(self.children_)
-        assert self.leaf_labels_.max() + 1 == len(self.max_weight_per_cc_) + len(
-            self.children_
-        )
+        if self.leaf_labels_.max() + 1 != len(self.max_weight_per_cc_) + len(self.children_):
+            raise RuntimeError(
+                "Wrong values were calculated for leaf_labels_, max_weight_per_cc_ and children_"
+            )
         # TODO: deduplicate this code with the setters below
         if self.__merge_threshold:
-            assert not self.__n_clusters
+            if self.__n_clusters:
+                raise ValueError("__n_clusters should be 0")
```

Comment:
mentioning internal names may be confusing. The message could say that the user shouldn't specify n_clusters and merge_threshold at the same time, they have to pick at most one.

Reply:
Ah, ok. ValueError made it look like the user passed a bad argument and we wanted to communicate that. If this is just an internal consistency check (the condition cannot happen, but the assert ensures we don't break this property in some future refactoring), let's keep it, but make it clear that's what it is.